src/sessions/PageManager.ts

```diff
@@ -44,11 +69,24 @@ export class PageManager {
     private recordInteraction: boolean;
 
     constructor(config: Config, record: RecordEvent) {
+        super();
         this.config = config;
         this.record = record;
         this.page = undefined;
         this.resumed = undefined;
         this.recordInteraction = false;
+
+        this.periodicCheckerId = undefined;
+        this.activityTimeoutCheckerId = undefined;
+        this.domMutationObserver = new MutationObserver(
+            this.domMutationCallback
+        );
+        this.requestBuffer = new Set();
+        this.PERIODIC_CHECKER_INTERVAL = 100;
+        this.ACTIVITY_TIMEOUT_LIMIT = this.config.spaActivityTimeoutLimit
+            ? this.config.spaActivityTimeoutLimit
+            : 1000;
```

Comment:
suggestion: Instead of using this member variable, [set a default value](https://github.com/aws-observability/aws-rum-web/blob/main/src/orchestration/Orchestration.ts#L104) for `spaActivityTimeoutLimit` in `Orchestration`.

Same for `PERIODIC_CHECKER_INTERVAL` if we add a config setting for that as well.

Reply:
Will reflect in next revision.

Reply:
Added both to config, to give more flexibility to the customer's application.